tests/test_url_handler.py

```diff
@@ -0,0 +1,696 @@
+    }]
+    results = url_handler.create_json_browse(project.project_slug, None, root)
+
+    assert json.loads(results) == expected
+
+
+@pytest.mark.parametrize('root, text, id', [
+    ('com,', '<a href="surt/http://(com,example">com,example</a>', 'com,example,'),
+    ('', 'com', 'com,'),
+])
+def test_create_json_browse_does_not_show_duplicates(root, text, id):
+    project = factories.ProjectFactory()
+    factories.SURTFactory.create_batch(
+        2,
+        url_project=project,
+        entity='http://www.example.com',
+        value='http://(com,example,www)'
+    )
+    expected = [{
+        'hasChildren': True,
+        'id': id,
+        'text': text
+    }]
+    results = url_handler.create_json_browse(project.project_slug, None, root)
+
+    assert json.loads(results) == expected
+
+
+def test_create_json_browse_valid_root_many_urls():
+    project = factories.ProjectFactory()
+    urls = factories.SURTFactory.create_batch(101, url_project=project)
+    root = 'com,'
+    results = url_handler.create_json_browse(project.project_slug, None, root)
+    expected = [
+        {
+            'hasChildren': True,
+            'id': root + x.value[x.value.find(root) + 4],
+            'text': x.value[x.value.find(root) + 4]
+        } for x in urls
+    ]
+
+    for result in json.loads(results):
+        assert result in expected
+
+    assert json.loads(results).sort() == expected.sort()
+
+
+def test_create_json_browse_project_does_not_exist():
+    slug = 'blah'
+    with pytest.raises(http.Http404):
+        url_handler.create_json_browse(slug, None, None)
+
+
+def test_create_json_browse_valid_root_without_matching_surt():
+    project = factories.ProjectFactory()
+    root = 'example'
+    assert url_handler.create_json_browse(project.project_slug, None, root) == '[]'
+
+
+def test_create_json_browse_empty_root_without_surt():
+    project = factories.ProjectFactory()
+    root = ''
+    assert url_handler.create_json_browse(project.project_slug, None, root) == '[]'
+
+
+@pytest.mark.xfail(reason='URLs are not being filtered by project.')
+def test_create_json_search_project_found():
+    project = factories.ProjectFactory()
+    expected_urls = factories.URLFactory.create_batch(10, url_project=project)
+    other_urls = factories.URLFactory.create_batch(10)
+    json_url_list = url_handler.create_json_search(project.project_slug)
+
+    for url in expected_urls:
+        assert url.entity in json_url_list
+
+    for url in other_urls:
+        assert url.entity not in json_url_list
+
+
+def test_create_json_search_project_not_found():
+    with pytest.raises(http.Http404):
+        url_handler.create_json_search('fake_slug')
+
+
+def test_create_url_list():
+    project = factories.ProjectFactory()
+    url = 'www.example.com'
+    surt = 'http://(com,example,www,)'
+    domain_surt = 'http://(com,example,'
+    rand_metadata = factories.URLFactory.create_batch(5, url_project=project, entity=url)
```

Comment:
Can we rename this to `metadata`? the fact that it is random is a detail of the factory.